conda_smithy/configure_feedstock.py

```diff
@@ -588,6 +589,17 @@ def _render_ci_provider(
             if ver:
                 os.environ["DEFAULT_LINUX_VERSION"] = ver
 
+        # detect if `compiler('cuda')` is used in meta.yaml,
+        # and set appropriate environment variable
+        with open(
+            os.path.join(forge_dir, forge_config["recipe_dir"], "meta.yaml")
+        ) as f:
+            meta_lines = f.readlines()
```

Comment:
Let's read the whole file into one string.

Reply:
OK, done.

Reply:
Actually, I went back to line-based processing, as otherwise, we'd do the wrong thing on recipes that contain something like
```
  - {{ compiler('c') }}
  # TODO: GPU support
  # - {{ compiler('cuda') }}
```